pkg/config/schema.go

```diff
@@ -36,35 +37,46 @@ var HandledTargets = []TargetKind{
-	Category       string                 `yaml:"category" json:"category"`
-	SuccessMessage string                 `yaml:"successMessage" json:"successMessage"`
-	FailureMessage string                 `yaml:"failureMessage" json:"failureMessage"`
-	Controllers    includeExcludeList     `yaml:"controllers" json:"controllers"`
-	Containers     includeExcludeList     `yaml:"containers" json:"containers"`
-	Target         TargetKind             `yaml:"target" json:"target"`
-	SchemaTarget   TargetKind             `yaml:"schemaTarget" json:"schemaTarget"`
-	Schema         map[string]interface{} `yaml:"schema" json:"schema"`
-	SchemaString   string                 `yaml:"jsonSchema" json:"jsonSchema"`
-	Validator      jsonschema.RootSchema  `yaml:"-"`
+	ID                      string                            `yaml:"id" json:"id"`
+	Category                string                            `yaml:"category" json:"category"`
+	SuccessMessage          string                            `yaml:"successMessage" json:"successMessage"`
+	FailureMessage          string                            `yaml:"failureMessage" json:"failureMessage"`
+	Controllers             includeExcludeList                `yaml:"controllers" json:"controllers"`
+	Containers              includeExcludeList                `yaml:"containers" json:"containers"`
+	Target                  TargetKind                        `yaml:"target" json:"target"`
+	SchemaTarget            TargetKind                        `yaml:"schemaTarget" json:"schemaTarget"`
+	Schema                  map[string]interface{}            `yaml:"schema" json:"schema"`
+	SchemaString            string                            `yaml:"jsonSchema" json:"jsonSchema"`
+	Validator               jsonschema.RootSchema             `yaml:"-"`
+	AdditionalSchemas       map[string]map[string]interface{} `yaml:"additionalSchemas" json:"additionalSchemas"`
+	AdditionalSchemaStrings map[string]string                 `yaml:"additionalSchemaStrings" json:"additionalSchemaStrings"`
+	AdditionalValidators    map[string]jsonschema.RootSchema  `yaml:"-"`
 }
 
 type resourceMinimum string
 type resourceMaximum string
 
-// ParseCheck parses a check from a byte array
-func ParseCheck(id string, rawBytes []byte) (SchemaCheck, error) {
-	reader := bytes.NewReader(rawBytes)
-	check := SchemaCheck{}
+func unmarshalYAMLOrJSON(raw []byte, dest interface{}) error {
+	reader := bytes.NewReader(raw)
 	d := k8sYaml.NewYAMLOrJSONDecoder(reader, 4096)
 	for {
-		if err := d.Decode(&check); err != nil {
+		if err := d.Decode(dest); err != nil {
 			if err == io.EOF {
 				break
 			}
-			return check, fmt.Errorf("Decoding schema check failed: %v", err)
+			return fmt.Errorf("Decoding schema check failed: %v", err)
 		}
 	}
+	return nil
+}
+
+// ParseCheck parses a check from a byte array
+func ParseCheck(id string, rawBytes []byte) (SchemaCheck, error) {
+	check := SchemaCheck{}
+	err := unmarshalYAMLOrJSON(rawBytes, &check)
```

Comment:
This will be a bit confusing because it will can accept both `check` and `&check` only to fail during runtime.

Reply:
Yeah agreed - I think go generally has this problem. E.g. [json.Unmarshal](https://golang.org/pkg/encoding/json/#Unmarshal) has the same behavior.

I wish there was a way to specify that you want a _pointer_ to an interface, and not just anything.